### ISSUE TITLE
Accessing hidden method Landroid/os/WorkSource

### ISSUE THREAD

**Issue by shokarta** [ceciletti/firebase-qt#4]
Hello @dantti,
after weeks of figuring this out, I was able to run it via QMake, android SDK used 34, minSDK 28, firebase c++ sdk is current 12.2.0
So I was able to run it and receive the device token :).
but when I send the push notification from firebase console, either directly to one device or to all, I receive the following "warnings"
```
W mtoro.appMobile: Accessing hidden method Landroid/os/WorkSource;->add(I)Z (unsupported,test-api, reflection, allowed)
W mtoro.appMobile: Accessing hidden method Landroid/os/WorkSource;->add(ILjava/lang/String;)Z (unsupported,test-api, reflection, allowed)
W mtoro.appMobile: Accessing hidden method Landroid/os/WorkSource;->get(I)I (unsupported, reflection, allowed)
W mtoro.appMobile: Accessing hidden method Landroid/os/WorkSource;->getName(I)Ljava/lang/String; (unsupported, reflection, allowed)
```
and then nothing else happens... no error, no notification, no console debug:
![image](https://github.com/user-attachments/assets/325bc264-e4ee-4ef1-9c52-62a4df974b83)
 - not sure if function "messageReceived" is even triggered by firebase anymore

when we got this far, would be sad to give up...
by your experience, or at least hunch, dont you have any idea where could be the problem or what to look up as a next step?

**Comment by dantti** [comment]
do you get the notification when the app is closed? If not then your server has issues, else when running the messageReceived should be triggered

**Comment by shokarta** [comment]
No I dont even when app is closed, but I dont see how this could be server issue. I send notifications directly from firebise console (so not via api myself). and once its sent, the app on device recognize incomping message, and prints the error from issue i reported above. So the apps know notification was received, it just does not know ow to process it (i think)

**Comment by shokarta** [comment]
sorry, my mistake... when app is closed, then i receive the notification just fine :)

**Comment by dantti** [comment]
right, well then it's an issue on the app side, could be even with the FB C++ sdk, but don't know what it is

**Comment by dantti** [comment]
Closing this one, as the cmake branch likely deals with the issues here
